x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -94,20 +100,37 @@ private void doExecuteForked(Task task, EsqlQueryRequest request, ActionListener
             EsqlPlugin.QUERY_RESULT_TRUNCATION_MAX_SIZE.get(settings)
         );
         String sessionId = sessionID(task);
-        planExecutor.esql(
-            request,
-            sessionId,
-            configuration,
-            listener.delegateFailureAndWrap(
-                (delegate, r) -> computeService.execute(sessionId, (CancellableTask) task, r, configuration, delegate.map(pages -> {
-                    List<ColumnInfo> columns = r.output()
-                        .stream()
-                        .map(c -> new ColumnInfo(c.qualifiedName(), EsqlDataTypes.outputType(c.dataType())))
-                        .toList();
-                    return new EsqlQueryResponse(columns, pages, request.columnar());
-                }))
-            )
-        );
+
+        try {
+            planExecutor.esql(
+                request,
+                sessionId,
+                configuration,
+                errorLoggingListener.delegateFailureAndWrap(
+                    (delegate, plan) -> computeService.execute(
+                        sessionId,
+                        (CancellableTask) task,
+                        plan,
+                        configuration,
+                        delegate.map(pages -> {
+                            List<ColumnInfo> columns = plan.output()
+                                .stream()
+                                .map(c -> new ColumnInfo(c.qualifiedName(), EsqlDataTypes.outputType(c.dataType())))
+                                .toList();
+                            EsqlQueryResponse response = new EsqlQueryResponse(columns, pages, request.columnar());
+
+                            return response;
+                        })
+                    )
+                )
+            );
+        } catch (RuntimeException e) {
```

Comment:
We should not catch the exception here; I don't think it will take effect. Instead of simply delegating the failure with `delegateFailureAndWrap`, I think we should wrap both the onResponse and onFailure methods of a listener.

Reply:
I am not sure if I understand correctly; in the current state of this PR, this catch block is necessary because I otherwise cannot log the query if it failed during parsing. I confirmed locally and on cloud deployment that this catch-block takes effect and logs the query in case of parsing errors.

I want to push a change though, which will remove this catch block and instead ensure that query parsing errors are correctly sent up the listeners rather than catching them separately, as Costin suggested.